test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -283,11 +284,12 @@ public class HubEndpointTests
 
             using (var client = new TestClient())
             {
-                // TestClient automatically writes negotiate, for this test we want to assume negotiate never gets sent
-                client.Connection.Transport.Input.TryRead(out var item);
-                client.Connection.Transport.Input.AdvanceTo(item.Buffer.End);
+                Task endPointTask = await client.ConnectAsync(endPoint, false);
```

Comment:
This test doesn't work, I don't think it has ever worked as intended (`NegotiateTimeout` doesn't do anything in the test), and I'm not sure how to fix it.

Instead of waiting on the server for content and being cancelled by the cancellation token, the test has always exited straight away.

Reply:
Checking for `NegotiationResponseMessage` below is why it doesn't work now. If negotiate was properly timing out then you would get a response saying so. Instead it doesn't send anything.

Reply:
The behavior we're testing is that the client never sends us a Negotiate request right? Couldn't we just call `OnConnectedAsync` directly for that rather than using our `ConnectAsync` helper? We should get a response after the timeout elapses containing the error message, right?

Reply:
The problem I'm observing is the server doesn't wait for a request message and so doesn't timeout. It immediately accepts that there is no content for the connection and exits.

Reply:
That seems wrong, it should be looping until either the pipeline completes, or there's enough information in the buffer to parse a negotiate message. In the case that `ReadAsync` never returns, the token should be firing. @davidfowl thoughts?

Reply:
Figured this test out